sampledb/reader.xsh

```diff
@@ -68,15 +69,19 @@ def upload_samples(host=None, db='sampleDB', collection='samples', key=None, use
         sdb = SampleDatabase('localhost:' + str(port), db, collection)
     else:
         sdb = SampleDatabase(host, db, collection)
-    uids = get_uid_from_qr()
-    known, unknown = find_uids(uids, sdb)
-    if len(known):
-        print('The following sample uids were found in the database:')
-    for uid in known:
-        print(uid)
-    write_sample_spreadsheet(unknown, sdb)
-    if key:
-        ssh -S sock -O exit @(server)
+    try:
+        uids = get_uid_from_qr()
+        known, unknown = find_uids(uids, sdb)
+        if len(known):
+            print('The following sample uids were found in the database:')
+        for uid in known:
+            print(uid)
+        write_sample_spreadsheet(unknown, sdb)
+    except Exception as e:
+        print(e)
+    finally:
+        if key:
+            ssh -S sock -O exit @(server)
```

Comment:
I think that `SampleDatabase` may be a good [context manager](https://www.python.org/dev/peps/pep-0343/).

In this way we can internalize the ssh tunnel/socket handling into the database itself and remove that from the rest of the code.

```python
with SampleDatabase(host, db, collection, key, user, port) as sdb:
    uids = get_uid_from_qr()
    known, unknown = find_uids(uids, sdb)
    ....
    write_sample_spreadsheet(unknown, sdb)
```